### ISSUE TITLE
RightCurly false-positives for multi-block continuation

### ISSUE THREAD

**Issue by Vampire** [checkstyle/checkstyle#6466]
https://checkstyle.org/config_blocks.html#RightCurly
https://checkstyle.org/property_types.html#rcurly `same` is default mode

```
/var/tmp $ javac Foo.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="RightCurly">
            <property name="tokens" value="CLASS_DEF"/>
            <property name="tokens" value="CTOR_DEF"/>
            <property name="tokens" value="METHOD_DEF"/>
        </module>
    </module>
</module>

/var/tmp $ cat Foo.java
class Foo {
    public Foo() {
        Thread t = new Thread(new Runnable() {
           @Override
           public void run() {
           } // <--
        });
    } // <--

    public void getId() {
    } // <--
} // <--

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-8.17-all.jar -c config.xml Foo.java
Starting audit...
[ERROR] /var/tmp/Foo.java:6:12: '}' at column 12 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally). [RightCurly]
[ERROR] /var/tmp/Foo.java:8:5: '}' at column 5 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally). [RightCurly]
[ERROR] /var/tmp/Foo.java:11:5: '}' at column 5 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally). [RightCurly]
[ERROR] /var/tmp/Foo.java:12:1: '}' at column 1 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally). [RightCurly]
Audit done.
Checkstyle ends with 4 errors.
```

Expected:
No violation
Default mode is `same` https://checkstyle.org/property_types.html#rcurly

From doc:
> Same - 	Works like alone_or_singleline but the brace should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else or try/catch/finally). If no next part of a multi-block statement present, brace must be alone on line. It also allows single-line format of multi-block statements.

**Comment by romani** [comment]
Please follow required format of bug report, that was defined in issue template

**Comment by Vampire** [comment]
Seriously?
You ask me to reformat my report that has all information you ask about in your issue template, just to fulfill some process?
You are aware that this way you scare away people from reporting bugs, aren't you?
I mean, if I would have provided insufficient information, I would totally understand and agree.
But this feels like pure chicane. :-(

**Comment by romani** [comment]
Yes, I know it looks like over demand, that format answer all possible questions and details that are required for us to make decision AND for person outside project take it an quickly  get to know how to reproduce.

>report that has all information you ask about in your issue template, just to fulfill some process?

I do not remember exactly, but I thought that I did not notice version that you use. I have no evidence of this now. Thanks a lot for update in description.
There were numerous examples in my practice that such format self resolve a lot issues as people even do not aware what version they use, ... .

>You are aware that this way you scare away people from reporting bugs, aren't you?

Yes, but in the same time do not want to scary away contributor of fix. Such format helps a lot.

>But this feels like pure chicane. :-(


Sorry about this. But this helps a lot to people who are going to fix the issue. And reporter of issue become understand that we are not addressing all issues of all plugins and ....... .
I appreciate your help with reporting issues.

**Comment by Vampire** [comment]
I'm sorry, it was 3am, maybe I was a bit too harsh.

> I do not remember exactly, but I thought that I did not notice version that you use.

That's indeed correct, forgot to mention that, but actually my first revised version also didn't have it, as I didn't change the commands from the template, just inserted my results, so the verison would have still be missing. I then just thought I'll be overly correct this time and replaced the commands too. :-D

> I have no evidence of this now.

You have, unless I delete them manually from the history, you can see old version of edited content on GitHub since quite some time, just click the `edited` string.

**Comment by Vampire** [comment]
The same problem also exists for `LITERAL_FOR`, `LITERAL_WHILE`, `STATIC_INIT` and `INSTANCE_INIT`.
